scanpy/tests/test_paga.py

```diff
@@ -1,61 +1,63 @@
 from functools import partial
-from importlib.util import find_spec
 from pathlib import Path
 
-from matplotlib import cm
+import pytest
 import numpy as np
+from matplotlib import cm
 
 import scanpy as sc
-from scanpy.tests._data._cached_datasets import pbmc68k_reduced, pbmc3k_processed
-import pytest
+from scanpy.testing._pytest.marks import needs_igraph
+
 
 HERE: Path = Path(__file__).parent
 ROOT = HERE / '_images'
 FIGS = HERE / 'figures'
 
 
-needs_igraph = pytest.mark.skipif(
-    not find_spec("igraph"),
-    reason="needs module `igraph` (`pip install python-igraph`)",
-)
-
-
 @pytest.fixture(scope="module")
-def pbmc():
-    pbmc = pbmc68k_reduced()
+def _pbmc(_pbmc68k_reduced):
+    pbmc = _pbmc68k_reduced.copy()
     sc.tl.paga(pbmc, groups='bulk_labels')
     pbmc.obs['cool_feature'] = pbmc[:, 'CST3'].X.squeeze()
     return pbmc
 
 
+@pytest.fixture
+def pbmc(_pbmc):
+    return _pbmc.copy()
+
+
+paga_plot_tests = [
+    pytest.param(sc.pl.paga, id=""),
+    pytest.param(partial(sc.pl.paga, color="CST3"), id="continuous"),
+    pytest.param(partial(sc.pl.paga, color="cool_feature"), id="continuous_obs"),
+    pytest.param(
+        partial(sc.pl.paga, color=['CST3', 'GATA2']), id="continuous_multiple"
+    ),
+    pytest.param(partial(sc.pl.paga_compare, legend_fontoutline=2), id="compare"),
+    pytest.param(
+        partial(sc.pl.paga_compare, color='CST3', legend_fontsize=5),
+        id="compare_continuous",
+        marks=pytest.mark.xfail(reason="expects .uns['paga']['pos']"),
+    ),
+    pytest.param(
+        partial(sc.pl.paga_compare, basis='X_pca', legend_fontweight='normal'),
+        id="compare_pca",
+    ),
+]
+
+
 @needs_igraph
 @pytest.mark.parametrize(
     "test_id,func",
-    [
-        ("master_paga", sc.pl.paga),
-        ("master_paga_continuous", partial(sc.pl.paga, color="CST3")),
-        ("master_paga_continuous_obs", partial(sc.pl.paga, color="cool_feature")),
-        (
-            "master_paga_continuous_multiple",
-            partial(sc.pl.paga, color=['CST3', 'GATA2']),
-        ),
-        ("master_paga_compare", partial(sc.pl.paga_compare, legend_fontoutline=2)),
-        (
-            "master_paga_compare_continuous",
-            partial(sc.pl.paga_compare, color='CST3', legend_fontsize=5),
-        ),
-        (
-            "master_paga_compare_pca",
-            partial(sc.pl.paga_compare, basis='X_pca', legend_fontweight='normal'),
-        ),
-    ],
+    [pytest.param(p.id, *p.values, id=p.id, marks=p.marks) for p in paga_plot_tests],
```

Comment:
so there is an association between a parametrized test name (e.g. `scanpy/tests/test_paga.py::test_paga_plots[continuous_obs]`) and the file name (e.g. `master_paga_continuous_obs.png`)

otherwise `test_paga_plots[func4]` fails and you have to count the list items to know which one that was

Reply:
Can you show me where this happens?

If I collect the tests on master, I see the `test_id` showing up in the collected name:

<details>
<summary> </summary>

```python
$ pytest -k test_paga -v
=================================================== test session starts ====================================================
platform darwin -- Python 3.9.16, pytest-7.2.0, pluggy-1.0.0 -- /usr/local/opt/python@3.9/bin/python3.9
cachedir: .pytest_cache
rootdir: /Users/isaac/github/scanpy, configfile: pyproject.toml, testpaths: scanpy/tests/
plugins: xdist-2.5.0, memray-1.4.0, assume-2.4.3, nunit-1.0.0, sphinx-0.5.0, profiling-1.7.0, anyio-3.6.2, forked-1.3.0, cov-3.0.0
collected 977 items / 965 deselected / 7 skipped / 12 selected                                                             

scanpy/tests/test_paga.py::test_paga_plots[master_paga-paga] PASSED                                                  [  8%]
scanpy/tests/test_paga.py::test_paga_plots[master_paga_continuous-func1] PASSED                                      [ 16%]
scanpy/tests/test_paga.py::test_paga_plots[master_paga_continuous_obs-func2] PASSED                                  [ 25%]
scanpy/tests/test_paga.py::test_paga_plots[master_paga_continuous_multiple-func3] PASSED                             [ 33%]
scanpy/tests/test_paga.py::test_paga_plots[master_paga_compare-func4] PASSED                                         [ 41%]
scanpy/tests/test_paga.py::test_paga_plots[master_paga_compare_continuous-func5] PASSED                              [ 50%]
scanpy/tests/test_paga.py::test_paga_plots[master_paga_compare_pca-func6] PASSED                                     [ 58%]
scanpy/tests/test_paga.py::test_paga_pie PASSED                                                                      [ 66%]
scanpy/tests/test_paga.py::test_paga_path PASSED                                                                     [ 75%]
scanpy/tests/test_paga.py::test_paga_compare PASSED                                                                  [ 83%]
scanpy/tests/test_paga.py::test_paga_positions_reproducible PASSED                                                   [ 91%]
```

</details>

Reply:
Hm, you’re right, they’re both there. Will revert